src/sql/src/parsers/create_parser.rs

```diff
@@ -705,6 +758,54 @@ ENGINE=mito";
         }
     }
 
+    #[test]
+    fn test_parse_create_table_with_timestamp_index() {
```

Comment:
Some cases I can think of:
- contains both `TIME INDEX` and `NOT NULL` in one line
- contains both `TIME INDEX` and `TIME INDEX (ts)`

Reply:
You are right. What the right behavior is if contains both TIME INDEX and TIME INDEX (ts)? Is this behavior the same as two duplicated `TIME INDEX (ts)` if no new syntax?

Reply:
>Is this behavior the same as two duplicated TIME INDEX (ts) if no new syntax?

Yes! The current behavior is 
```
ERROR 1815 (HY000): Failed to execute query: CREATE TABLE monitor (
  host_id    INT,
  idc        STRING,
  ts         TIMESTAMP,
  TIME INDEX (ts),
  TIME INDEX (ts),
  PRIMARY KEY (host),
)
ENGINE=mito, source: Invalid SQL, error: must have one and only one TimeIndex columns
```
and this is how MySQL handles it:
```sql
mysql> create table test_pk3(id int not null primary key, val int, primary key(id));
ERROR 1068 (42000): Multiple primary key defined
```

Reply:
I checkout develop branch, and add test case below, found duplicated `TIME INDEX (ts)` is valid in `ParserContext::create_with_dialect`, Does that mean this error is not handled in this level of abstraction?

<img width="800" alt="Screen Shot 2022-12-06 at 21 49 35" src="https://user-images.githubusercontent.com/5784607/205930418-1c46ccbe-0a19-4c17-a528-54bfc3b9ee3d.png">

https://github.com/SSebo/greptimedb/commit/bfbcf0f0291df139ccc53b5f007a97c439902850

Reply:
Yes, currently the parser only checks errors like grammar. This kind of "logical conflict" is checked in the next step (convert the parsed SQL AST to inner gRPC request at https://github.com/GreptimeTeam/greptimedb/blob/c5b0d2431f4095c7a429e84124a594e4fd56df78/src/frontend/src/expr_factory.rs#L85

Reply:
I mean should I test duplicated time index for new syntax?

Reply:
>I mean should I test duplicated time index for new syntax? It's Eq to normal syntax for next step in create_parser.rs.

In this parser module we can only test on the grammar level, and left logic check to downstream like before.

>Should ‘ts TIMESTAMP NOT NULL TIME INDEX’ and ‘ts TIMESTAMP TIME INDEX NOT NULL’ be valid syntax?

I think they are the same, and both are valid. Just like `NOT NULL PRIMARY KEY` and `PRIMARY KEY NOT NULL` are equal in MySQL. This might bring some complicity to implementation, thanks in advance 🫣

Reply:
Does ‘need not to write NOT NULL, this is implied by TIME INDEX keywords’ mean 'NOT NULL' is always added for new syntax? No matter if 'NOT NULL' is added explicit or not?

Reply:
Yes. `ts TIME INDEX` and `ts TIME INDEX NOT NULL` are the same -- `TIME INDEX` can be considered as a "grammar candy" which implies this column is "not null" and "timestamp index".